kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterScheduler.scala

```diff
@@ -70,121 +72,50 @@ private[spark] class KubernetesClusterScheduler(conf: SparkConf)
     val kubernetesHost = "k8s://" + client.getMasterUrl().getHost()
     logInfo("Using as kubernetes-master: " + kubernetesHost.toString())
 
-    var annotationMap = Map("pod.beta.kubernetes.io/init-containers" -> raw"""[
-                {
-                    "name": "client-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/client.jar", "$clientJarUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "distro-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/spark.tgz", "$sparkDistUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "setup",
-                    "image": "$sparkDriverImage",
-                    "command": ["./install.sh"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        },
-                        {
-                            "name": "opt",
-                            "mountPath": "/opt"
-                        }
-                    ]
-                }
-            ]""")
-
-
     val labelMap = Map("type" -> "spark-driver")
     val pod = new PodBuilder()
       .withNewMetadata()
       .withLabels(labelMap.asJava)
       .withName(driverName)
-      .withAnnotations(annotationMap.asJava)
       .endMetadata()
       .withNewSpec()
       .withRestartPolicy("OnFailure")
+      .withServiceAccount(serviceAccountName)
       .addNewContainer()
       .withName("spark-driver")
       .withImage(sparkDriverImage)
       .withImagePullPolicy("Always")
-      .withCommand("/opt/spark/bin/spark-submit")
-      .withArgs(s"--class=${args.userClass}",
+      .withCommand(s"/opt/driver.sh")
+      .withArgs(s"$clientJarUri",
+                s"--class=${args.userClass}",
                 s"--master=$kubernetesHost",
                 s"--executor-memory=${driverDescription.mem}",
                 s"--conf=spark.executor.jar=$clientJarUri",
                 s"--conf=spark.executor.instances=$instances",
-                s"--conf=spark.kubernetes.namespace=${getNamespace()}",
+                s"--conf=spark.kubernetes.namespace=$nameSpace",
                 s"--conf=spark.kubernetes.driver.image=$sparkDriverImage",
-                s"--conf=spark.kubernetes.distribution.uri=$sparkDistUri",
-                "/work-dir/client.jar",
+                "/opt/spark/kubernetes/client.jar",
```

Comment:
To elaborate, I don't think it would be very hard to support an image with user app jars pre-installed, nor would it be hard to download a spark-distro.jar and unpack it with annotations, it just seems non-idiomatic to me.   (And the annotations support is currently wonky.   My openshift install didn't recognize it, and _failed silently_)